### DISCUSSION TITLE
isFiller cell API

### DISCUSSION THREAD

**Discussion by vikrg23** [The-OpenROAD-Project/OpenROAD#5806] [General]
I am checking if an instance is a filler or not using the isFiller API in OpenROAD. But the API returns 0 even if the cell is filler.
On debugging that I found the API definition in dbMaster.cpp file, which checks the type of master cell and returns 1 only if the master type is CORE_SPACER. In my design, all the filler cells have their type as CORE. So, I am wondering if its a design issue or is there an accurate way to check if a cell is filler.

I have attached a testcase for this issue. After extracting the file contents, go into the testcase directory and source the runme.tcl file in OpenROAD. 
[isfiller_testcase.zip](https://github.com/user-attachments/files/17143326/isfiller_testcase.zip)

**Comment by maliberty** [comment]
Use https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/platforms/nangate45/lef/NangateOpenCellLibrary.macro.mod.lef instead.

**Comment by rovinski** [comment]
To elaborate, there is no way for OpenROAD or any other tool to know if something is a filler unless it is properly specified in the library (i.e. CORE SPACER). You could make a guess based on the fact that it only has power and ground pins and no other pins, but it could still be a decap cell in that case.
